k8s/upgrades/0.8.0-0.8.1/cstor_pool_update.sh

```diff
@@ -0,0 +1,120 @@
+       rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+       echo "Deployment upgrade strategy set as recreate"
+    else
+       echo "Deployment upgrade strategy was already set as recreate"
+    fi
+}
+
+##Checking the version of OpenEBS using image in deployment ####
+function checkVersion() {
+	ns=$1
+	check_deply=$2
+	openebs_version=$(kubectl get deploy $check_deply -n $ns\
+        -o jsonpath="{.spec.template.spec.containers[0].image}" | cut -d ':' -f 2)
+
+	if [ $openebs_version != "0.8.0" ]; then
+    		echo "Current replica version is not 0.8.0";exit 1;
+	fi
+
+}
+
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+spc=$1
+ns=$2
+pool_cnt=0
+
+# Get the list of pool deployments for given SPC, delimited by ':'
+pool_deploys=`kubectl get deploy -n $ns \
+ -l app=cstor-pool,openebs.io/storage-pool-claim=$spc \
+ -o jsonpath="{range .items[*]}{@.metadata.name}:{end}"`
+
+echo "Patching Pool Deployment upgrade strategy as recreate"
+for pool_dep in `echo $pool_deploys | tr ":" " "`; do
+    if [ $pool_cnt -eq 0 ]; then
+    	checkVersion $ns $pool_dep
+    fi
+    setDeploymentRecreateStrategy $ns $pool_dep
+    pool_cnt=$((pool_cnt+1))
+done
+
+
+echo "Patching Pool Deployment with new image"
+for pool_dep in `echo $pool_deploys | tr ":" " "`; do
+    pool_rs=$(kubectl get rs -n openebs \
+ -o jsonpath="{range .items[?(@.metadata.ownerReferences[0].name=='$pool_dep')]}{@.metadata.name}{end}")
+    echo "$pool_dep -> rs is $pool_rs"
+
+    #fetch the csp_uuid
+    csp_uuid="";csp_uuid=`kubectl get csp -n $ns $pool_dep -o jsonpath="{.metadata.uid}"`
+    echo "$pool_dep -> csp uuid is $csp_uuid"
+    if [  -z "$csp_uuid" ];
+    then
+       echo "Error: Unable to fetch csp uuid";
+       exit 1
+    fi
+
+    sed "s/@csp_uuid[^ \"]*/$csp_uuid/g" cstor-pool-patch.tpl.json > cstor-pool-patch.json
+
+    kubectl patch deployment --namespace $ns $pool_dep -p "$(cat cstor-pool-patch.json)"
+    rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+    rollout_status=$(kubectl rollout status --namespace $ns deployment/$pool_dep)
+    rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+    then echo "ERROR: $rc"; exit; fi
+    kubectl delete rs $pool_rs --namespace $ns
+    rm cstor-pool-patch.json
+done
+
+## Cross check whether the pods are in running status or not if not wait untill it comes to running####
+while [ true ]
+do
+  cnt_aft_upg=$(kubectl get po -n $ns -l app=cstor-pool --no-headers | grep Running | wc -l)
+	if [ $cnt_aft_upg -eq $pool_cnt ]
+	then
+			break
+	else
+		  sleep 5
+	fi
+done
+
+running_pool_pods=$(kubectl get po -n $ns -l app=cstor-pool -o jsonpath='{.items[?(@.status.phase=="Running")].metadata.name}'| tr " " "\n")
+
+## Setting the quorum enabled in the new pool pods ###
+for pool_pod in $running_pool_pods
+do
+	pool_name=$(kubectl exec $pool_pod -n $ns -c cstor-pool-mgmt -- zpool list | awk 'NR==2 {print $1 }')
```

Comment:
you have some other way of getting poolname right..?

Reply:
I will update it. i.e by cstor-UID of the corresponding CSP resource?